### PR TITLE
[eclipse/xtext#1644] use java platform

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1644] use java platform
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
updated to use bom defined in gradle too.
needs re-review

**Comment by cdietrich** [comment]
we might also have to adapt https://ci.eclipse.org/xtext/job/releng/job/check-versions/

File: org.eclipse.xtext.dev-bom/build.gradle [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                description = 'Xtext Third Party Dependencies (Bill of Materials)'
+                url = 'https://www.eclipse.org/Xtext/'
+                licenses {
+                    license {
+                        name = 'Eclipse Public License, Version 1.0'
+                        url = 'http://www.eclipse.org/legal/epl-v10.html'
+                    }
+                }
+                scm {
+                    connection = "scm:git:git@github.com:eclipse/${rootProject.name}.git"
```
Comment (kthoms) [inline org.eclipse.xtext.dev-bom/build.gradle]: git URL does not work for anonymous user (permission denied), use HTTPS URL:
`connection = "scm:git:https://github.com/eclipse/${rootProject.name}.git`

File: org.eclipse.xtext.dev-bom/build.gradle [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+publishing {
+    publications {
+        LocalMavenRepo(MavenPublication) {
+            from components.javaPlatform
+            artifactId "xtext-dev-bom"
+            pom {
+                name = 'Eclipse Xtext Developer BOM'
+                description = 'Xtext Third Party Dependencies (Bill of Materials)'
+                url = 'https://www.eclipse.org/Xtext/'
+                licenses {
```
Comment (kthoms) [inline org.eclipse.xtext.dev-bom/build.gradle]: The property definition
```
<project.build.sourceEncoding>ISO-8859-1</project.build.sourceEncoding>
```
is missing in the resulting POM. Please add

```
                properties = [
                    "project.build.sourceEncoding": "ISO-8859-1"
                ]
```

File: org.eclipse.xtext.dev-bom/build.gradle (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.dev-bom/build.gradle]: please open a different issue. this affects all gradle stuff

File: org.eclipse.xtext.dev-bom/build.gradle (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.dev-bom/build.gradle]: done

File: org.eclipse.xtext.dev-bom/build.gradle (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtext.dev-bom/build.gradle]: https://github.com/eclipse/xtext/issues/1652

**Pull request merged**